crates/rover-client/src/operations/persisted_queries/publish/types.rs

```diff
@@ -152,3 +236,71 @@ impl PersistedQueriesOperationCounts {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn it_can_read_relay_manifest() {
+        let relay_manifest = r#"{
+      "ed145403db84d192c3f2f44eaa9bc6f9": "query NewsfeedQuery {\n  topStory {\n    title\n    summary\n    poster {\n      __typename\n      name\n      profilePicture {\n        url\n      }\n      id\n    }\n    thumbnail {\n      url\n    }\n    id\n  }\n}\n"
+    }"#;
+
+        let relay_manifest: RelayPersistedQueryManifest =
+            serde_json::from_str(relay_manifest).expect("could not read relay manifest");
+        assert_eq!(relay_manifest.operations.len(), 1);
+    }
+
+    #[test]
+    fn it_can_convert_relay_manifest_to_apollo_manifest() {
+        let id = "ed145403db84d192c3f2f44eaa9bc6f9";
+        let body = "query NewsfeedQuery {\n  topStory {\n    title\n    summary\n    poster {\n      __typename\n      name\n      profilePicture {\n        url\n      }\n      id\n    }\n    thumbnail {\n      url\n    }\n    id\n  }\n}\n";
+        let relay_manifest = serde_json::json!({id: body}).to_string();
+
+        let relay_manifest: RelayPersistedQueryManifest =
+            serde_json::from_str(&relay_manifest).expect("could not read relay manifest");
+        let apollo_manifest: ApolloPersistedQueryManifest = relay_manifest.try_into().unwrap();
+        assert_eq!(
+            apollo_manifest.operations[0],
+            PersistedQueryOperation {
+                name: "NewsfeedQuery".to_string(),
+                r#type: PersistedQueryOperationType::Query,
+                id: id.to_string(),
+                body: body.to_string()
+            }
+        );
+    }
+
+    #[test]
+    fn relay_manifest_with_anonymous_operations_fails() {
```

Comment:
Or, taking another look, tests for these would be great but I don't think either would pass at the moment. (RC shouldn't ever create such a document, but we should handle it well if it does!)

Reply:
While the tests you added were good, they weren't the cases I was thinking about. I'm thinking about the case where an *individual body* contains 0 or multiple operations.

Basically, the whole reason that this code is more complex than the Apollo manifest case is that we have to figure out the name and type of "the operation" in each document. So the edge cases we need to think about are the ones where "the operation" is not a well defined concept.  Having the number of operation definitions in a document be not equal to 1 is a reason that this could be not well defined.